File: ProgrammingSkills/MatrixDiagonalSum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        result = 0
        for i in range(len(mat)):
            result += mat[i][i] + mat[i][~i]
        if (len(mat) % 2):
            result -= mat[int(i/2)][int(i/2)]
            
        return result

File: JSProblems/matrixDiagonalSum.js
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function(mat) {
    let result = 0;
    let lastPos = mat[0].length;
    let mid = Math.floor( lastPos / 2 );

    for(let i  = 0; i<lastPos; i++){
        result += mat[i][i];
        result += mat[lastPos-1-i][i];            
    }
    if( lastPos % 2 == 1 ){        
        result -= mat[mid][mid];
    }
    return result;  
};
